set.mm

```diff
@@ -419491,6 +419515,22 @@ the sequence p(0) p(1) ... p(n-1) of the vertices in a closed walk p(0)
                                  { w e. ( ClWWalks ` g ) | ( # ` w ) = n } ) $.
   $}
 
+  ${
+    $d g n v w $.
+    $( Define the set of all closed walks a graph ` g ` , anchored at a fixed
+       vertex ` v ` (i.e., a walk starting and ending at the fixed vertex
+       ` v ` , also called "a closed walk on vertex ` v ` ") and having a fixed
+       length ` n ` , as words over the set of vertices.  Such a word
+       corresponds to the sequence v=p(0) p(1) ... p(n-1) of the vertices in a
+       closed walk p(0) e(f(1)) p(1) e(f(2)) ... p(n-1) e(f(n)) p(n)=p(0)=v as
+       defined in ~ df-clwlks .  The set ` ( ( v ( ClWWalksOnN `` g ) n ) `
+       corresponds to the set of "walks from v to v of length n" in a statement
+       of [Huneke] p. 2.  (Contributed by AV, 24-Feb-2022.) $)
+    df-clwwlksonn $a |- ClWWalksOnN = ( g e. _V
+                    |-> ( v e. ( Vtx ` g ) , n e. NN0
+                          |-> { w e. ( n ClWWalksN g ) | ( w ` 0 ) = v } ) ) $.
+  $}
+
```

Comment:
It looks like this definition is used much later ? If so, maybe put it later ?

Reply:
I usually look for definition right at the start of a chapter, just after the new constants are introduced, so maybe putting the definition later would make them a bit harder to find?

Reply:
This is a general question about how sections with more than one definition should be structured. In my opinion, all definitions should be at the beginning, then the theorems should follow, grouped by theitr usage of the definitions (first group: only first defiition is used, second group: the first and the second definition used, etc.). This makes sence at least if the definitions are based on each other (as in the case of closed walks as words).

Maybe some basic theorems, however, should follow immediately after the definitions. If this is preferred, the section "Closed walks as words" could be structured as follows: 

  $c ClWWalks $.
  $c ClWWalksN $.
  $c ClWWalksOnN $.

  cclwwlks $a class ClWWalks $.
  cclwwlksn $a class ClWWalksN $.
  cclwwlksonn $a class ClWWalksOnN $.

    df-clwwlks $a |- ClWWalks = ... $.
    df-clwwlksn $a |- ClWWalksN = ... ClWWalks ... $.
    df-clwwlksonn $a |- ClWWalksOnN = ... ClWWalksN ... $.

    clwwlks $p |- ( ClWWalks ` G ) = ... $=
    isclwwlks $p |- ( W e. ( ClWWalks ` G ) <-> ... $=

    clwwlksn $p |- ( N e. NN -> ( N ClWWalksN G ) = ... $=
    isclwwlksn $p |- ( N e. NN -> ( W e. ( N ClWWalksN G ) <-> ... $=

    clwwlksonnmpt2 $p |- ( ClWWalksOnN ` G ) = ... $=
    clwwlksonn $p |- ( ( X e. V /\ N e. NN0 ) -> ( X ( ClWWalksOnN ` G ) N ) = ... $=
    isclwwlksonn $p |- ( ( X e. V /\ N e. NN0 ) -> ( W e. ( X ( ClWWalksOnN ` G ) N ) <-> ... $=

    [theorems using ClWWalks only]

    [theorems using ClWWalks and/or ClWWalksN]

    [theorems using ClWWalks and/or ClWWalksN and/or ClWWalksOnN]

If acceptable, I would move up the theorems ~ clwwlksonnmpt2 , ~clwwlksonn and ~ isclwwlksonn.

Reply:
Since @tirix merged this PR already (thank you), I could move up the theorems ~ clwwlksonnmpt2 , ~clwwlksonn and ~ isclwwlksonn in my next PR. Maybe we should discuss my proposal for structuring sections with more than one definition separately (shall I open an issue for it?) and add it as convention if it is accepted.

Reply:
Then I suggest you have three section (either add two sections at the same level, or subdivide the section into three subsections), titled e.g. "Closed walks", "Closed walks based at a vertex", "Closed walks based at a vertex of a given length".  It is easier to be introduced to a new notion one at a time.

Reply:
> Then I suggest you have three sections (either add two sections at the same level, or subdivide the section into three subsections), titled e.g. "Closed walks", "Closed walks based at a vertex", "Closed walks based at a vertex of a given length". It is easier to be introduced to a new notion one at a time.

So you propose that we should have at most one definition in a section in general? With our four header levels thisould maybe be possible, but I can remember that Norm was not a friend of such an approach. If we want to follow this suggestion, we have to start with section "Logical disjunction and conjunction", which should be split into two sections "Logical disjunction" and "Logical conjunction" and so forth. We should have a broader discussion on this...

Reply:
This would have my preference in most cases, but this does not have to be a rule.  My comment above was about one specific case. As for the broader discussion... maybe, but I do not have the time to launch it now. So do as you prefer.